### ISSUE TITLE
should NamedTempFile implement Read/Write on shared references?

### ISSUE THREAD

**Issue by oconnor663** [Stebalien/tempfile#24]
I think the standard `File` implements `Read` and `Write` on `&'a File` (in addition to plan old `File`) to allow writing through shared references. (I could be understanding this totally wrong, but it [seems to work](http://is.gd/Zr0mTq) :)) Would it make sense for `NamedTempFile` to do the same?

**Comment by Stebalien** [comment]
Given that `NamedTempFile` derefs to `File` is that really necessary?

**Comment by oconnor663** [comment]
Oh cool, I missed that. I'm not familiar with how method resolution works, but I get a compiler when I try to write with a shared ref to a NamedTempFile. I have to use something like `(&**f).write_all(...)` to work around it. I wonder if _removing_ the existing Read and Write implementations would allow all of that to fall through to File? Anyway, this crate is great, thanks for writing it.

**Comment by Stebalien** [comment]
I've implemented several traits (Read/Write/AsRaw*) on on NamedTempFile directly so one can write `Box::new(NamedTempFile::new()) as Box<SomeTraits>`.

While I don't think adding the requested implementations would make reading/writing to an immutable NamedTempFile any more ergonomic (`(&*tmpfile)` versus `(&tmpfile)`), it certainly makes these methods more discoverable and mirrors `File` so I've decided to implement them anyways (version 2.1.2). Thanks for the suggestion.
